src/utils/GlobalContext.tsx

```diff
@@ -0,0 +1,196 @@
+      };
+
+    case 'setFilter':
+      const newFilter = action.payload;
+
+      return {
+        ...state,
+        filter: newFilter,
+        filteredTodos: filterTodos(LocalStorage.get(), newFilter),
+      };
+
+    case 'addTodo':
+      const newPost: Todo = {
+        id: getNewId(),
+        title: action.payload,
+        completed: false,
+      };
+
+      LocalStorage.set([...LocalStorage.get(), newPost]);
+
+      return {
+        ...state,
+        ...updateAllTodos(),
+      };
+
+    case 'toggleAll':
+      const currentToToggleAll = LocalStorage.get();
+      let toggledTodos: Todo[] = [];
+
+      if (currentToToggleAll.every(todo => todo.completed)) {
+        toggledTodos = currentToToggleAll.map((todo: Todo) => ({
+          ...todo,
+          completed: false,
+        }));
+      } else {
+        toggledTodos = currentToToggleAll.map((todo: Todo) => ({
+          ...todo,
+          completed: true,
+        }));
+      }
+
+      LocalStorage.set(toggledTodos);
+
+      return {
+        ...state,
+        ...updateAllTodos(),
+      };
+
+    case 'editTodo':
+      const currentToDel = LocalStorage.get();
+      const { id, title } = action.payload;
+      const editTodoIndex = currentToDel.findIndex(el => el.id === id);
+
+      currentToDel[editTodoIndex].title = title;
+
+      LocalStorage.set(currentToDel);
+
+      return {
+        ...state,
+        ...updateAllTodos(),
+      };
+
+    case 'toggleTodo':
+      const currentToToggle = LocalStorage.get();
+      const toogleTodoIndex = currentToToggle.findIndex(
+        el => el.id === action.payload,
+      );
+
+      currentToToggle[toogleTodoIndex].completed =
+        !currentToToggle[toogleTodoIndex].completed;
+
+      LocalStorage.set(currentToToggle);
+
+      return {
+        ...state,
+        ...updateAllTodos(),
+      };
+
+    case 'deleteTodo':
+      const newTodos = LocalStorage.get().filter(
+        (todo: Todo) => todo.id !== action.payload,
+      );
+
+      LocalStorage.set(newTodos);
+
+      return {
+        ...state,
+        ...updateAllTodos(),
+      };
+
+    case 'deleteAllCompleted':
+      const allNotCompleted = LocalStorage.get().filter(
+        todo => !todo.completed,
+      );
+
+      LocalStorage.set(allNotCompleted);
+
+      return {
+        ...state,
+        ...updateAllTodos(),
+      };
+
+    default:
+      return { ...state };
+  }
+}
+
+const initialState: State = {
+  todos: [],
+  filteredTodos: [],
+  filter: Filters.all,
+};
+
+export const StateContext = React.createContext(initialState);
+
+export const DispatchContext = React.createContext<React.Dispatch<Action>>(
+  () => {},
+);
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const GlobalProvider: React.FC<Props> = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    dispatch({ type: 'getTodos' });
+    dispatch({ type: 'getFilteredTodos' });
```

Comment:
it's a bit uncommon and better to remove filteredTodos from the reducer. General rule - no need to store value in the state if it can be computed from other state values. Compute filtered todos in place you need it based on todos and filter

So you will get rid of dispatching this action here